### PR TITLE
New Query Runner: Apache Pinot

### PR DESCRIPTION
**Pull request by justrp**
## What type of PR is this? (check all applicable)

- [x] New Query Runner (Data Source)

## Description

Query runner for http://pinot.apache.org/

## Related Tickets & Documents

https://github.com/apache/incubator-pinot/issues/6297

### CODE REVIEW

**Comment by etadelta222** [comment]
Hi @kravets-levko , has this been abandoned?

**Comment by justrp** [comment]
oh, wow, I almost forgot about this one: it's been almost a year)
rebased from current master branch

**Comment by susodapop** [comment]
Hey @justrp and @etadelta222 thanks so much for the ping. We just introduced our new process for reviews which includes a new PR template you can see [here](https://raw.githubusercontent.com/getredash/redash/master/.github/PULL_REQUEST_TEMPLATE.md). Would you mind updating the PR description? Really all we need to see is the applicable versions of pinot for which this query runner will work.

I'm looking forward to getting this reviewed and merged ASAP!

File: redash/query_runner/pinot.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+logger = logging.getLogger(__name__)
+
+
+PINOT_TYPES_MAPPING = {
+    "INT": TYPE_INTEGER,
+    "LONG": TYPE_INTEGER,
+    "FLOAT": TYPE_FLOAT,
+    "DOUBLE": TYPE_FLOAT,
+    "STRING": TYPE_STRING,
+    "BYTES": TYPE_STRING,
```
Comment (xiangfu0) [inline redash/query_runner/pinot.py:23]: Can you add more types here, especially the TIMESTAMP

File: requirements_all_ds.txt [hunk 2]
```diff
@@ -42,3 +42,4 @@ xlrd==2.0.1
 openpyxl==3.0.7
 firebolt-sdk
 pandas==1.3.4
+pinotdb==0.3.3
```
Comment (xiangfu0) [inline requirements_all_ds.txt:45]: 0.3.11 is the latest version :p

**Comment by etadelta222** [comment]
Hi @susodapop , is the information provided by Xiangfu sufficient?

**Comment by justrp** [comment]
in case if  you wait for me to address these changes - unfortunately, can't provide clear ETA - got a very busy time now :(
hopefully, will have some time at week after the next one - but can't predict for sure

**Comment by susodapop** [comment]
Another contributor has opened #5724 which duplicates this PR. Since @justrp is currently busy, I'm closing this one for now but have asked the author of 5724 to incorporate @xiangfu0's feedback into their effort.

**Pull request closed without merging**